### ISSUE TITLE
ethminer fails when internet is down?

### ISSUE THREAD

**Issue by kkkrackpot** [ethereum-mining/ethminer#386]
Hi,

Recent ethminer from master branch on Linux with Nvidia 1060.

It seems ethminer fails when internet is down or there's some other network erros:
```
  ℹ  03:19:41|stratum   Received new job #a8738e3a
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::system::system_error> >'
  what():  write: Connection reset by peer
```

Before I've also noticed that it sometimes fails after _waiting 3 seconds to reconnect_ too (but I don't have an example at the moment).

Not sure what exactly is wrong here.

Adding failover server doesn't help.

Regards,
Alex

**Comment by Jacxz** [comment]
Hi, I'm not sure this is exactly the same issue,
but at least I'm also on the latest master (0.13.0.dev0), Linux and Nvidia 1060.

What I'm experiencing is that the pool stops accepting submitted solutions for some reason? Resulting in ethminer trying to submit stale shares until throwing an exception and crashing.

The same boost exception is thrown but the output/reason is a little different "what():  write: Broken pipe".

>   ℹ  20:26:51|CUDA1     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:26:51|CUDA1       Nonce: 0x1b9ebcd9325930a4
>   ℹ  20:26:51|stratum    B-) Submitted and accepted.
>   ℹ  20:27:06|stratum   Received new job #f1701c57
>   ℹ  20:27:06|CUDA0     set work; seed: #89c875b9, target:  #0000000112e0
>   ℹ  20:27:06|CUDA1     set work; seed: #89c875b9, target:  #0000000112e0
>   ℹ  20:27:39|CUDA1     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:27:39|CUDA1       Nonce: 0x92503d625242c1b0
>   ℹ  20:27:48|CUDA3     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:27:48|CUDA3       Nonce: 0x8fd2ad6152bc1386
>   ✘  20:27:48|CUDA3     Submitting stale solution.
>   ℹ  20:28:03|CUDA5     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:28:03|CUDA5       Nonce: 0x525d673fda93acd8
>   ✘  20:28:03|CUDA5     Submitting stale solution.
>   ℹ  20:28:06|CUDA4     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:28:06|CUDA4       Nonce: 0xb215c81e230f56eb
>   ✘  20:28:06|CUDA4     Submitting stale solution.
>   ℹ  20:29:09|CUDA0     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:29:09|CUDA0       Nonce: 0x73e4443933f37a8e
>   ℹ  20:29:42|CUDA1     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:29:42|CUDA1       Nonce: 0x92503d62fa622111
>   ℹ  20:29:52|CUDA0     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:29:52|CUDA0       Nonce: 0x73e444396defc216
>   ℹ  20:30:01|CUDA5     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:30:01|CUDA5       Nonce: 0x525d67407b454cab
>   ✘  20:30:01|CUDA5     Submitting stale solution.
>   ℹ  20:30:15|CUDA3     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:30:15|CUDA3       Nonce: 0x8fd2ad621e785117
>   ✘  20:30:15|CUDA3     Submitting stale solution.
>   ℹ  20:31:11|CUDA3     Solution found; Submitting to eu1.ethermine.org ...
>   ℹ  20:31:11|CUDA3       Nonce: 0x8fd2ad626c77f43b
>   ✘  20:31:11|CUDA3     Submitting stale solution.
>   ℹ  20:31:50|CUDA2     set work; seed: #89c875b9, target:  #0000000112e0
>   ℹ  20:31:51|CUDA3     set work; seed: #89c875b9, target:  #0000000112e0
>   ℹ  20:31:51|CUDA4     set work; seed: #89c875b9, target:  #0000000112e0
>   ℹ  20:31:51|CUDA5     set work; seed: #89c875b9, target:  #0000000112e0
>   ✘  20:31:51|stratum   Handle response failed: Broken pipe
>   ✘  20:31:51|stratum   Handle response failed: Broken pipe
>   ✘  20:31:51|stratum   Handle response failed: Broken pipe
>   ✘  20:31:51|stratum   Handle response failed: Broken pipe
>   ✘  20:31:51|stratum   Handle response failed: Broken pipe
>   ℹ  20:31:51|stratum   Received new job #311cc626
> terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::system::system_error> >'
>   what():  write: Broken pipe
> ./miner.sh: line 24:  2566 Aborted                 (core dumped) ~/ethereum-mining/ethminer/build/ethminer/ethminer --farm-recheck 2000 -v 1 -RH -U -S "eu1.ethermine.org:4444" -FS "eu2.ethermine.org:4444" -O "$MY_ADDRESS.$MY_RIG"

This is my first post here, hopefully I'm correct in using this issue instead of creating a new one :)
_(btw what markdown is used to get that scroll-able code/quote?)_

**Comment by derubm** [comment]
@Jacxz just use pastebin :P

**Comment by Jacxz** [comment]
I keep getting the Broken pipe error :(
Maybe I should mention that the linux rig is running of a USB flash drive

**Comment by derubm** [comment]
did you allready try those values?

` -SC, --stratum-client <n>  Stratum client version. Defaults to 1 (async client). Use 2 to use the new synchronous client.
    -SP, --stratum-protocol <n> Choose which stratum protocol to use:
        0: official stratum spec: ethpool, ethermine, coinotron, mph, nanopool (default)
        1: eth-proxy compatible: dwarfpool, f2pool, nanopool
        2: EthereumStratum/1.0.0: nicehash`

**Comment by Jacxz** [comment]
I've just found issue #171 which is exactly about my problem, so will take further comments there instead.
Thanks for the input @derubm (I'll try with -SC 2 and report back).

**Comment by DeadManWalkingTO** [comment]
Test launch command:

`ethminer.exe -RH -U -S eu1.ethermine.org:4444 -FS us1.ethermine.org:4444 -O 0x7013275311fc37ccc1e40193D75086293eCb43A4.test`

Try with latest Ethminer version and feedback please.
Thank you!

**Comment by AndreaLanfranchi** [comment]
Addressed in #1027
